### PR TITLE
Folder path is now displayed when hovering a tab. (#6808)

### LINKED ISSUE #6808: Same folder name in tab title
### What's the Problem?

I often have multiple folders with the same name opened in different tabs, and it's hard to distinguish them.

### Solution/Idea

Show a brief preview of the path directing to the folder behind its name on the tab. Similar to VSCode. (only if two folders with the same name are opened).

### Alternatives

?

### Priorities

| Capability | Priority |
| :---------- | :------- |
| This proposal will allow developers to accomplish W | Must |
| This proposal will allow end users to accomplish X | Should |
| This proposal will allow developers to accomplish Y | Could |
| This proposal will allow end users to accomplish Z | Won't |

### Files Version

v2.0.34.0

### Windows Version

Windows 11 Home Insider Preview 22494.1000

### Additional comment

_No response_

### PR DESCRIPTION
**Pull request by QuaintMako**
**Resolved / Related Issues**
Items resolved / related issues by this PR.
- Closes #6808 

**Details of Changes**
Add details of changes here.
- Added a HoverDisplayText field and property in TabItem.
- Binded said property in xaml of Horizontal and Vertical Multitasking Controls.
- Updated the function GetSelectedTabInfoAsync().

**Validation**
How did you test these changes?
- [ X ] Built and ran the app

_Note:_ INavigationControlItem defines HoverDisplayText as well, should we extract it and define a new interface, something like IHoverable ?

**Screenshots**
6808_proof

### CODE REVIEW

File: src/Files.Uwp/UserControls/MultitaskingControl/HorizontalMultitaskingControl.xaml [hunk 1]
```diff
@@ -590,7 +590,7 @@
                         Header="{x:Bind Header, Mode=OneWay}"
                         IconSource="{x:Bind IconSource, Mode=OneWay}"
                         Style="{StaticResource Local.TabViewItem}"
-                        ToolTipService.ToolTip="{x:Null}" />
+                        ToolTipService.ToolTip="{Binding HoverDisplayText}" />
```
Comment (yaichenbaum) [inline src/Files.Uwp/UserControls/MultitaskingControl/HorizontalMultitaskingControl.xaml:593]: Does x:Bind work here?

File: src/Files.Uwp/UserControls/MultitaskingControl/TabItem/TabItem.cs [hunk 2]
```diff
@@ -29,6 +29,14 @@ public string Description
             set => SetProperty(ref description, value);
         }
 
+        private string hoverDisplayText;
+
+        public string HoverDisplayText
```
Comment (yaichenbaum) [inline src/Files.Uwp/UserControls/MultitaskingControl/TabItem/TabItem.cs:34]: Not a requirement but we're trying to add proper comments for new properties to better define what they're used for.

**Comment by yaichenbaum** [comment]
> Note: INavigationControlItem defines HoverDisplayText as well, should we extract it and define a new interface, something like IHoverable ?

Good idea, do you want to do that as part of this PR?

File: src/Files.Uwp/UserControls/MultitaskingControl/HorizontalMultitaskingControl.xaml (hunk 1 above)
Comment (QuaintMako) [reply src/Files.Uwp/UserControls/MultitaskingControl/HorizontalMultitaskingControl.xaml:593]: Will try it tonight.

File: src/Files.Uwp/UserControls/MultitaskingControl/TabItem/TabItem.cs (hunk 2 above)
Comment (QuaintMako) [reply src/Files.Uwp/UserControls/MultitaskingControl/TabItem/TabItem.cs:34]: Alright, I will add this in a new commit!

**Comment by QuaintMako** [comment]
Sure

> > Note: INavigationControlItem defines HoverDisplayText as well, should we extract it and define a new interface, something like IHoverable ?
> 
> Good idea, do you want to do that as part of this PR?

Sure thing, I will add this. Is there a namespace adapted for such interface?

**Comment by yaichenbaum** [comment]
@d2dyno1

File: src/Files.Uwp/UserControls/MultitaskingControl/HorizontalMultitaskingControl.xaml (hunk 1 above)
Comment (QuaintMako) [reply src/Files.Uwp/UserControls/MultitaskingControl/HorizontalMultitaskingControl.xaml]: Doesn't seems to work like so @yair100.
I'm not really experienced with xaml though, so if someone else wants to give it a shot, please be my guest!

Comment (yaichenbaum) [inline src/Files.Uwp/UserControls/MultitaskingControl/HorizontalMultitaskingControl.xaml:593]:
```suggestion
                        ToolTipService.ToolTip="{x:Bind HoverDisplayText, Mode=OneWay}" />
```
Can you try this?

Comment (QuaintMako) [reply src/Files.Uwp/UserControls/MultitaskingControl/HorizontalMultitaskingControl.xaml:593]: It works perfectly! I'm closing this side of the conversation, thanks for your help! :)

Comment (QuaintMako) [reply src/Files.Uwp/UserControls/MultitaskingControl/HorizontalMultitaskingControl.xaml:593]: Resolved in another line of comment.

File: src/Files.Uwp/UserControls/MultitaskingControl/TabItem/TabItem.cs (hunk 2 above)
Comment (QuaintMako) [reply src/Files.Uwp/UserControls/MultitaskingControl/TabItem/TabItem.cs:38]: It has been added. Was it the expected format?

**Comment by QuaintMako** [comment]
I have already made the refactoring into an interface.
Now, it needs to be validated name-wise and placement-wise (Most likely wrong for now.)

File: src/Files.Uwp/UserControls/MultitaskingControl/TabItem/TabItem.cs (hunk 2 above)
Comment (yaichenbaum) [reply src/Files.Uwp/UserControls/MultitaskingControl/TabItem/TabItem.cs:38]: Perfect

File: src/Files.Uwp/Helpers/HoverDisplay/IHoverableDisplay.cs [hunk 3]
```diff
@@ -0,0 +1,7 @@
+﻿namespace Files.Uwp.Helpers.HoverDisplay
+{
+    public  interface IHoverableDisplay
```
Comment (yaichenbaum) [inline src/Files.Uwp/Helpers/HoverDisplay/IHoverableDisplay.cs:3]: We don't really have a good place for this right now, but we can easily move this when we do so we can leave this as is.

**Review by yaichenbaum — DISMISSED** [review]
LGTM
@d2dyno1 do you want to look this over?

File: src/Files.Uwp/Helpers/HoverDisplay/IHoverableDisplay.cs (hunk 3 above)
Comment (d2dyno1) [inline src/Files.Uwp/Helpers/HoverDisplay/IHoverableDisplay.cs:3]: Why do we need this interface? By the looks of things, it's not used anywhere except for the implementation

Comment (yaichenbaum) [reply src/Files.Uwp/Helpers/HoverDisplay/IHoverableDisplay.cs:3]: We use this on the sidebar as well

Comment (yaichenbaum) [reply src/Files.Uwp/Helpers/HoverDisplay/IHoverableDisplay.cs:3]: @QuaintMako I was wrong, I'm realizing that the sidebar is pretty different from the tab item, and it doesn't make sense to share this interface. We'll probably create an interface for the current directory at some point, but I think we can revert this change.

Comment (QuaintMako) [reply src/Files.Uwp/Helpers/HoverDisplay/IHoverableDisplay.cs:3]: Aight! Thanks for review it guys! I'm rollbacking the change!

**Review by d2dyno1 — APPROVED** [review]

**Pull request merged**